packages/core/src/common/utils.ts

```diff
@@ -71,11 +71,25 @@ export function getDisplayName(ComponentClass: React.ComponentType | INamed) {
     return (ComponentClass as React.ComponentType).displayName || (ComponentClass as INamed).name || "Unknown";
 }
 
+/**
+ * Returns true if the given JSX element matches the given component type.
+ *
+ * NOTE: This function only checks equality of `displayName` for performance and
+ * to tolerate multiple minor versions of a component being included in one
+ * application bundle.
+ * @param element JSX element in question
+ * @param ComponentType desired component type of element
+ */
 export function isElementOfType<P = {}>(
     element: any,
-    ComponentClass: React.ComponentType<P>,
+    ComponentType: React.ComponentType<P>,
 ): element is React.ReactElement<P> {
-    return element != null && element.type === React.createElement(ComponentClass).type;
+    return (
+        element != null &&
+        element.type != null &&
+        element.type.displayName != null &&
+        element.type.displayName === ComponentType.displayName
+    );
```

Comment:
I'm looking forward to [optional-chaining](https://github.com/tc39/proposal-optional-chaining)